app/Http/Controllers/DataControllers/SourceController.php

```diff
@@ -25,11 +25,26 @@ public function addSourceSubmit(Request $request) {
         if (empty($request->source))
             return redirect()->route('sourceAdd')->with(['alert' => 'danger', 'alertMessage' => 'Please enter a source name!']);
 
+        $is_source = $request->boolean('is_source');
+        $is_destination = $request->boolean('is_destination');
+        $is_fixture = $request->boolean('is_fixture');
+
         $source = new ReuseNode();
         $source->node_name = $request->source;
-        $source->is_source = true;
-        $source->is_destination = false;
-        $source->is_fixture = false;
+
+        if($is_source === true)
+            $source->is_source = true;
+        else
+            $source->is_source = false;
+        if($is_destination === true)
+            $source->is_destination = true;
+        else
+            $source->is_destination = false;
+        if($is_fixture === true)
+            $source->is_fixture = true;
+        else
+            $source->is_fixture = false;
+
```

Comment:
Could we ditch the if statements and just assign the Boolean values directly?
E.g. `$source->is_source = $is_source` or even `$source->is_source = $request->boolean('is_source')`
The, Laravel function `$request->boolean()` can only return `true` or `false` so that should be fine.

Reply:
true!